lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -318,6 +334,19 @@ public void OnNext(IDriverMessage value)
             }
         }
 
+        /// <summary>
+        /// Propagates the IDriverConnection message to the Handler as specified by the Task.
+        /// </summary>
+        internal void HandleDriverConnectionMessage(IDriverConnectionMessage message)
+        {
+            if (_driverConnectionMessageHandler.Value == null)
+            {
+                return;
+            }
+
+            _driverConnectionMessageHandler.Value.OnNext(message);
```

Comment:
This is an Interface implemented by the user. Users can potentially (though not very likely) implement something that's more costly.